packages/utils/with-result.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+// This will take a promise and always use `resolve` to return a result in the format `[err, result]`.
```

Comment:
This could probably be removed from here if you put it into the documentation spot I highlighted above...